ls1intum/Artemis
# Fix Bug messing up changes in the MC-Quiz-Questions Markdown Editor

### Motivation and Context


The Markdown editor of MC questions in quizzes could not be used, it should weird behavior during typing. The issue seems to be critical, as new questions could not be well written (if at all) at the current state.
Closes #3277 

### Description
By comparing the `DragAndDropQuestionEditComponent` (where the editor is working) with the `MultipleChoiceQuestionEditComponent`(where the bug was observed), I found out, that the order of functions called when changes in the editor were detected was different.
As far as I can see, the proposed fix is related to code that was written some longer time ago and I can not explain, why my previous assumption, that the bug was caused by recently changed code, i.e. a recent regression, does not hold - it should have been detected earlier in the case is was introduced by the code that I change with my PR...
Nevertheless, the editor is no working as expected.

### Steps for Testing

The related issue has a screencast attached, showing how it should not be anymore :) 
1. Select a course in the course overview (as instructor)
2. Click on 'Exercises'
3. Scroll down to 'Quiz Exercises.'
4. Click 'Create Quiz'
5. Scroll down to 'Questions'
6. Click green button 'Add Multiple-Choice Question'
7. Try to change the pre-filled content
8. Save the adapted question
